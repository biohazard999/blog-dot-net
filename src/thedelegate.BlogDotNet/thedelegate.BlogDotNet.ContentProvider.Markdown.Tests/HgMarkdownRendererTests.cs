using System;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using FakeItEasy;
using Mercurial;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MSTestExtensions;

namespace thedelegate.BlogDotNet.ContentProvider.Markdown.Tests
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class HgMarkdownRendererTests
    {
        private string _RepoPath;
        private Repository _Repo;

        [TestInitialize]
        public void TestInitialize()
        {
            _RepoPath = Path.Combine(Path.GetTempPath(), Guid.Empty.ToString());
            Directory.CreateDirectory(_RepoPath);

            _Repo = new Repository(_RepoPath);
            _Repo.Clone("https://biohazard999@bitbucket.org/biohazard999/x2.delegate.at", new CloneCommand());
        }

        [TestCleanup]
        public void TestCleanup()
        {
            _Repo = null;
            SilentExecute(() => Directory.Delete(_RepoPath, true));
        }

        [TestMethod]
        [TestCategory("Integration")]
        public void Render_FirstPost_2013()
        {
            var path = Path.Combine(_RepoPath, "2013", "1-markdown-centric-blog", "markdown-centric-blog.md");

            IContentExecutor contentExecutor = new MarkDownContentExecutor(path);

            IContentProvider provider = contentExecutor.FillProvider(A.Fake<IContentProvider>());

            Approvals.Verify(provider.Content);
        }

        private void SilentExecute(Action action)
        {
            try
            {
                action();
            }
            // ReSharper disable once EmptyGeneralCatchClause
            catch
            {
               
            }
        }

        [TestMethod]
        public void Constuctor_IfFileIsMissing_ThrowsFileNotFoundException()
        {
            var filename = Path.Combine(Path.GetTempPath(), Guid.Empty.ToString());
     
            ExceptionAssert.Throws<FileNotFoundException>(() => new MarkDownContentExecutor(filename), 
                "The File was not found");
        }

    }
}
