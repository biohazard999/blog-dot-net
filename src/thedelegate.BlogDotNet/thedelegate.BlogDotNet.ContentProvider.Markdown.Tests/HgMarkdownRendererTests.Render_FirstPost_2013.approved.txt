<h2>Markdown centric blog engine</h2>
<p>There is an idea I've got a few years ago when i started blogging on our <a href="http://blog.paraoffice.at">companies developer blog</a> to use developer tools to drive a blog without having to force to use the browser to provide content for a blog.</p>
<p>I like <a href="http://daringfireball.net/projects/markdown/">markdown</a> cause the ease of use and the ability to merge within a VCS with ease. Also the possibility to provide semantic HTML without loosing the human readability.</p>
<p>The upcoming posts will document my progress to find/develop the <em>perfect</em> developer blog engine. </p>
<p>Stay tuned :)</p>
